### ISSUE TITLE
clicking on "empty modified file system cache" in System information cause SI crash

### ISSUE THREAD

**Issue by MagicAndre1981** [winsiderss/systeminformer#2085]
### Brief description of your issue

clicking on " empty modified file system" in System information cause SI crash

https://github.com/winsiderss/systeminformer/assets/8274816/2e057e6d-bca7-4bdd-9574-5d364ecf2f06

### Steps to reproduce (optional)

- start SI on Windows 10 22h2
- open System Information
- click on Memory->Empty>modified file system cache

### Expected behavior (optional)

no crash

### Actual behavior (optional)

  ```

KEY_VALUES_STRING: 1

    Key  : AV.Dereference
    Value: NullClassPtr

    Key  : AV.Fault
    Value: Read

  Key  : WER.OS.Branch
    Value: vb_release

    Key  : WER.OS.Version
    Value: 10.0.19041.1

    Key  : WER.Process.Version
    Value: 3.0.12187.7660

*** Stack trace for last set context - .thread/.cxr resets it
 # Child-SP          RetAddr               Call Site
00 000000a6`598fe8c0 00007ff7`1436d12b     SystemInformer!__crt_stdio_output::output_processor<wchar_t,__crt_stdio_output::string_output_adapter<wchar_t>,__crt_stdio_output::standard_base<wchar_t,__crt_stdio_output::string_output_adapter<wchar_t> > >::type_case_s_compute_narrow_string_length+0x31 [minkernel\crts\ucrt\inc\corecrt_internal_stdio_output.h @ 2361] 
01 000000a6`598fe8f0 00007ff7`1436c4c7     SystemInformer!__crt_stdio_output::output_processor<wchar_t,__crt_stdio_output::string_output_adapter<wchar_t>,__crt_stdio_output::standard_base<wchar_t,__crt_stdio_output::string_output_adapter<wchar_t> > >::type_case_s+0x87 [minkernel\crts\ucrt\inc\corecrt_internal_stdio_output.h @ 2343] 
02 000000a6`598fe920 00007ff7`1436ae8d     SystemInformer!__crt_stdio_output::output_processor<wchar_t,__crt_stdio_output::string_output_adapter<wchar_t>,__crt_stdio_output::standard_base<wchar_t,__crt_stdio_output::string_output_adapter<wchar_t> > >::state_case_type+0x12b [minkernel\crts\ucrt\inc\corecrt_internal_stdio_output.h @ 2066] 
03 000000a6`598fe9a0 00007ff7`1436dbbe     SystemInformer!__crt_stdio_output::output_processor<wchar_t,__crt_stdio_output::string_output_adapter<wchar_t>,__crt_stdio_output::standard_base<wchar_t,__crt_stdio_output::string_output_adapter<wchar_t> > >::process+0x139 [minkernel\crts\ucrt\inc\corecrt_internal_stdio_output.h @ 1704] 
04 (Inline Function) --------`--------     SystemInformer!common_vsprintf+0xe7 [minkernel\crts\ucrt\src\appcrt\stdio\output.cpp @ 170] 
05 000000a6`598fe9e0 00007ff7`142c2cb6     SystemInformer!__stdio_common_vswprintf+0x156 [minkernel\crts\ucrt\src\appcrt\stdio\output.cpp @ 256] 
06 (Inline Function) --------`--------     SystemInformer!_vscwprintf_l+0x26 [C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\ucrt\corecrt_wstdio.h @ 1340] 
07 (Inline Function) --------`--------     SystemInformer!_vscwprintf+0x26 [C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\ucrt\corecrt_wstdio.h @ 1358] 
08 000000a6`598fef30 00007ff7`142b26d1     SystemInformer!PhFormatString_V+0x36 [phlib\basesup.c @ 3076] 
09 000000a6`598fef80 00007ff7`141ec498     SystemInformer!PhShowMessage2+0x71 [phlib\util.c @ 716] 
0a (Inline Function) --------`--------     SystemInformer!PhDereferenceObject+0xd [phlib\ref.c @ 229] 
0b 000000a6`598ff0a0 00007ff7`1428d951     SystemInformer!PhShowMemoryListCommand+0x478 [SystemInformer\memlists.c @ 555] 
0c 000000a6`598ff180 00007ff9`530931e0     SystemInformer!PhSipMemoryPanelDialogProc+0xb1 [SystemInformer\sysscmem.c @ 505] 
0d 000000a6`598ff1c0 00007ff9`53092982     user32!UserCallDlgProcCheckWow+0x144
0e 000000a6`598ff2a0 00007ff9`53092896     user32!DefDlgProcWorker+0xd2
0f 000000a6`598ff360 00007ff9`5308ef75     user32!DefDlgProcW+0x36
10 000000a6`598ff3a0 00007ff9`5308dfcb     user32!UserCallWinProcCheckWow+0x515
11 000000a6`598ff530 00007ff9`5308d81d     user32!SendMessageWorker+0x70b
12 000000a6`598ff5d0 00007ff9`3b612457     user32!SendMessageW+0x18d
13 000000a6`598ff630 00007ff9`3b6220e0     comctl32!Button_ReleaseCapture+0xbb
14 000000a6`598ff660 00007ff9`5308ef75     comctl32!Button_WndProc+0x800
15 000000a6`598ff790 00007ff9`5308e69d     user32!UserCallWinProcCheckWow+0x515
16 000000a6`598ff920 00007ff9`5308bc66     user32!DispatchMessageWorker+0x49d
17 000000a6`598ff9a0 00007ff7`142819f8     user32!IsDialogMessageW+0x336
18 000000a6`598ffa30 00007ff7`142c0435     SystemInformer!PhSipSysInfoThreadStart+0x178 [SystemInformer\sysinfo.c @ 146] 
19 000000a6`598ffce0 00007ff9`53667344     SystemInformer!PhpBaseThreadStart+0x75 [phlib\basesup.c @ 188] 
1a 000000a6`598ffd20 00007ff9`550826b1     kernel32!BaseThreadInitThunk+0x14
1b 000000a6`598ffd50 00000000`00000000     ntdll!RtlUserThreadStart+0x21


FAULTING_SOURCE_LINE:  minkernel\crts\ucrt\inc\corecrt_internal_stdio_output.h

FAULTING_SOURCE_FILE:  minkernel\crts\ucrt\inc\corecrt_internal_stdio_output.h

FAULTING_SOURCE_LINE_NUMBER:  2361

FAULTING_SOURCE_CODE:  
No source found for 'minkernel\crts\ucrt\inc\corecrt_internal_stdio_output.h'


SYMBOL_NAME:  SystemInformer+31

MODULE_NAME: SystemInformer

IMAGE_NAME:  SystemInformer.exe

0:013> .frame 0n8;dv /t /v
08 000000a6`598fef30 00007ff7`142b26d1     SystemInformer!PhFormatString_V+0x36 [phlib\basesup.c @ 3076] 
@r14              wchar_t * Format = 0x00000238`022f8060 "Unable to flush volume file cache. Der Datenträger ist schreibgeschützt und kann nicht beschrieben werden. Entfernen Sie den Schreibschutz vom Datenträger %hs in Laufwerk %hs."
@rbp              char * ArgPtr = 0x000000a6`598ff0c8 "???"
```

I have an external USB drive enclosure with 3 USB3.2 Gen1 ports and an SATA port with an instaled SSD attached, but turned power for the drive off, so I see the drive in explorer but it is related to it as I see this text above in the dump:

![image](https://github.com/winsiderss/systeminformer/assets/8274816/c3667d48-8870-4c8f-8878-1aba1746e430)

### Environment (optional)

```shell
.7660 canary
Windows 10 22H2 19045.4412
```

**Comment by MagicAndre1981** [comment]
Under 17763 the command fails, too, but here I see an error message and get no crash as in 19045

![image](https://github.com/winsiderss/systeminformer/assets/8274816/3b4a8c8a-a971-4a5b-8290-5e7fe443aedc)

**Comment by MagicAndre1981** [comment]
Under 22631 it works fine:

![SI_success](https://github.com/winsiderss/systeminformer/assets/8274816/0adf3204-ea85-4a7a-bef9-06763bedfe61)

**Comment by ToddIndry37** [comment]
Got the same issue here. It crashed when I try to clear the cache of modified pages.

![image](https://github.com/winsiderss/systeminformer/assets/88038185/43de456e-354c-41df-b424-5fa9be848858)

**Comment by MagicAndre1981** [comment]
fixed with Version 3.1.24184.1. Thanks
